include/drivers/flash.h

```diff
@@ -161,12 +161,31 @@ static inline int z_impl_flash_erase(struct device *dev, off_t offset,
  *  @brief  Enable or disable write protection for a flash memory
  *
  *  This API is required to be called before the invocation of write or erase
- *  API. Please note that on some flash components, the write protection is
+ *  API. Behaviour at calls to @ref flash_write @ref flash_erase without
+ *  disabling the flash protection prior is undefined.
+ *  Usage Example:
+ *  @code
+ *   flash_write_protection_set(flash_dev, false);
+ *   flash_erase(flash_dev, page_offset, page_size);
+ *
+ *   flash_write_protection_set(flash_dev, false);
+ *   flash_write(flash_dev, offset, data, sizeof(data));
+ *
+ *   flash_write_protection_set(flash_dev, true); // enable is recommended
+ *  @endcode
+ *
+ *  Please note that on some flash components, the write protection is
  *  automatically turned on again by the device after the completion of each
- *  write or erase calls. Therefore, on those flash parts, write protection needs
- *  to be disabled before each invocation of the write or erase API. Please refer
- *  to the sub-driver API or the data sheet of the flash component to get details
- *  on the write protection behavior.
+ *  write or erase calls. Therefore, on those flash parts, write protection
```

Comment:
"Therefore, ~on those flash parts,~ _for portability across flash implementations_ write protection..."

The original (existing) text suggests the extra calls are optional.  That's true only if the underlying flash implementation happens to allow their absence, which is not a portable assumption.

Reply:
sure